src/dimension_range.c

```diff
@@ -0,0 +1,1127 @@
+/*
```

Comment:
Maybe this file should go in `src/ts_catalog`?

Reply:
Hmmm, many files like `dimension.c` and `dimension_slice.c` are in the `src` directory.

Reply:
@erimatnor shifted the files into `src/ts_catalog`